Src/IronPython/Modules/unicodedata.cs

```diff
@@ -137,7 +137,7 @@ public class UCD {
             public string unidata_version { get; private set; }
 
             public string lookup(string name) {
-                return char.ConvertFromUtf32(nameLookup[name]);
+                return char.ConvertFromUtf32(nameLookup[name.ToUpperInvariant()]);
```

Comment:
I think that indeed it would be a better option. Although `StringComparer.OrdinalIgnoreCase` is not nearly as optimized as the regular string equality method, `ToUpperInvariant` will create a new string object just for the sake of comparison.

I have the whole `unicodedata` module (and associated data files and scripts) on my to-do list but at a very low priority. Since I am not going to work on it any time soon, I will submit an Issue about it. It will be a "good first issue" candidate.

Reply:
`StringComparer.OrdinalIgnoreCase` lands in this PR though.